### PR TITLE
[MISC] Update search_cfg::max_error_substitution to recent config design changes.

### PR DESCRIPTION
**Pull request by rrahn**
part of seqan/product_backlog#217

### CODE REVIEW

**Review by wvandertoorn — APPROVED** [review]

File: include/seqan3/search/configuration/max_error.hpp [hunk 1]
```diff
@@ -50,11 +50,34 @@ class max_error_total : public pipeable_config_element<max_error_total, std::var
  * ### Example
  * \include test/snippet/search/configuration_error.cpp
  */
-class max_error_substitution : public pipeable_config_element<max_error_substitution,
-                                                              std::variant<error_count,
-                                                              error_rate>>
+struct max_error_substitution : public pipeable_config_element<max_error_substitution>
 {
-public:
+    //!\brief  The number of substitution errors allowed for the search.
```
Comment (smehringer) [inline include/seqan3/search/configuration/max_error.hpp:55]:
```suggestion
    //!\brief The number of substitution errors or the substitution error rate allowed for the search.
```

File: include/seqan3/search/configuration/max_error.hpp [hunk 2]
```diff
@@ -50,11 +50,34 @@ class max_error_total : public pipeable_config_element<max_error_total, std::var
...
+     */
+    max_error_substitution() = default; //!< Defaulted.
+    max_error_substitution(max_error_substitution const &) = default; //!< Defaulted.
+    max_error_substitution(max_error_substitution &&) = default; //!< Defaulted.
+    max_error_substitution & operator=(max_error_substitution const &) = default; //!< Defaulted.
+    max_error_substitution & operator=(max_error_substitution &&) = default; //!< Defaulted.
+    ~max_error_substitution() = default; //!< Defaulted.
+
+    /*!\brief Initialises the substitution error with the given seqan3::search_cfg::error_count.
+     * \param[in] error The error count used for the search.
```
Comment (smehringer) [inline include/seqan3/search/configuration/max_error.hpp:69]:
```suggestion
     * \param[in] error The number of substitution errors allowed in the search.
```

File: include/seqan3/search/configuration/max_error.hpp [hunk 3]
```diff
@@ -50,11 +50,34 @@ class max_error_total : public pipeable_config_element<max_error_total, std::var
...
+     */
+    explicit max_error_substitution(error_count error) : error{std::move(error)}
+    {}
+
+    /*!\brief Initialises the substitution error with the given seqan3::search_cfg::error_rate.
+     * \param[in] error The error rate used for the search.
```
Comment (smehringer) [inline include/seqan3/search/configuration/max_error.hpp:75]:
```suggestion
     * \param[in] error The substitution error rate allowed in the search.
```

**Review by smehringer — APPROVED** [review]
only dox

**Pull request merged**